nflow-engine/src/main/java/io/nflow/engine/service/WorkflowDefinitionService.java

```diff
@@ -54,7 +56,9 @@ public WorkflowDefinitionService(WorkflowDefinitionDao workflowDefinitionDao, En
    * @return List of workflow definitions.
    */
   public List<AbstractWorkflowDefinition<? extends WorkflowState>> getWorkflowDefinitions() {
-    return new ArrayList<>(workflowDefinitions.values());
+    synchronized (workflowDefinitions) {
```

Comment:
Why pessimize the get operations?
- If this method is not called often then it does not matter and the new code might be a bit simpler to read.
- If this method is called often then we should precalculate the `workflowDefinitions.values()` in an unmodifiableList and return that

Reply:
Optiomized for get